<?xml version="1.0"?>
<launch>
    <!-- MAVROS posix SITL environment launch script -->
    <!-- launches MAVROS, PX4 SITL, Gazebo environment, and spawns vehicle -->

    <arg name="mundo" default="pruebas"/>

    <!-- vehicle model and world -->
    <arg name="gui" default="false"/>
    <arg name="vehicle" default="hitl_iris_gimbal"/>
    <arg name="world" default="$(find offbnode)/worlds/$(arg mundo).world"/>
    <!-- MAVROS configs -->
    <arg name="gcs_url" default="udp://:18570@localhost:14550"/>

    <!-- Gazebo sim -->
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>

<!-- nodos -->
    <include file="$(find offbnode)/launch/mavros_posix_sitl.launch">
        <arg name="world" value="$(arg world)"/>
        <arg name="paused" default="$(arg paused)"/>
        <arg name="verbose" default="$(arg verbose)"/>
        <arg name="vehicle" value="$(arg vehicle)"/>
        <arg name="gcs_url" value="$(arg gcs_url)"/>
        <arg name="gui" default="$(arg gui)"/>
    </include>

<!-- evaluar resultado -->
    <node type="evaluar_recorrido.py" name="evaluar_recorrido" pkg="offbnode" output="screen" />

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find offbnode)/rviz/nav.rviz" />
    <node type="model_to_rviz.py" name="model_to_rviz" pkg="offbnode" output="screen" args="$(arg mundo)" />


</launch>
